src/nanotron/config/config.py

```diff
@@ -317,6 +330,22 @@ def __post_init__(self):
                 self.tokens.train_steps - self.optimizer.learning_rate_scheduler.lr_warmup_steps
             )
 
+        if self.data_stages is not None:
+            names = [stage.name for stage in self.data_stages]
+            training_steps = [stage.start_training_step for stage in self.data_stages]
+            assert all(
+                stage.start_training_step == 1 for stage in self.data_stages
```

Comment:
Do we start at 1 not 0?

Reply:
because we actually start from 1

> self.start_iteration_step + 1

https://github.com/huggingface/nanotron/blob/07caff1940609f09096cb4b0fd555441799ea252/src/nanotron/trainer.py#L289